### PR TITLE
Set orig data & last saved data when models are creted by collection loading

### PR DESCRIPTION
**Pull request by Sekiphp**
# Not tested yet

### Fixed Issues (if relevant)

1. Fixes OpenMage/magento-lts#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

File: lib/Varien/Data/Collection/Db.php [hunk 1]
```diff
@@ -586,11 +586,14 @@ public function load($printQuery = false, $logQuery = false)
 
         if (is_array($data)) {
             foreach ($data as $row) {
+                /** @var Mage_Core_Model_Abstract $item */
```
Comment (kiatng) [inline lib/Varien/Data/Collection/Db.php:589]: `$item` can be `Varien_Object`, if this is the case `$item->setOrigData();` is not applicable,

File: lib/Varien/Data/Collection/Db.php [hunk 2]
```diff
@@ -586,11 +586,14 @@ public function load($printQuery = false, $logQuery = false)
...
                 $item = $this->getNewEmptyItem();
                 if ($this->getIdFieldName()) {
                     $item->setIdFieldName($this->getIdFieldName());
                 }
                 $item->addData($row);
+                $item->setOrigData();
+                $item->setLastSavedData();
```
Comment (kiatng) [inline lib/Varien/Data/Collection/Db.php:596]: I cannot find any reference to the method `setLastSavedData()`. What does it do?

File: lib/Varien/Data/Collection/Db.php (hunk 1 above)
Comment (FreeWall) [reply lib/Varien/Data/Collection/Db.php:589]: > `$item` can be `Varien_Object`, if this is the case `$item->setOrigData();` is not applicable,

`Varien_Object` has `setOrigData` method

Comment (FreeWall) [inline lib/Varien/Data/Collection/Db.php:589]:
```suggestion
```
`$item` does not have to be `Mage_Core_Model_Abstract`, object class is determined by `$_itemObjectClass`

**Review by FreeWall — CHANGES_REQUESTED** [review]

File: lib/Varien/Data/Collection/Db.php (hunk 1 above)
Comment (kiatng) [reply lib/Varien/Data/Collection/Db.php:589]: My bad.

**Comment by fballiano** [comment]
I'm copying here the complete code of this PR:

```
From 4ce5c1b665b18ae1324acd2b4f57350e50432f01 Mon Sep 17 00:00:00 2001
From: Lubos Hubacek <lubos.hubacek@ateli.cz>
Date: Mon, 25 Apr 2022 00:41:38 +0200
Subject: [PATCH] Set orig data, set last saved data when models are creted by
 collection loading

---
 lib/Varien/Data/Collection/Db.php | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/Varien/Data/Collection/Db.php b/lib/Varien/Data/Collection/Db.php
index 15490075c78..05aa078066a 100644
--- a/lib/Varien/Data/Collection/Db.php
+++ b/lib/Varien/Data/Collection/Db.php
@@ -586,11 +586,14 @@ public function load($printQuery = false, $logQuery = false)
 
         if (is_array($data)) {
             foreach ($data as $row) {
+                /** @var Mage_Core_Model_Abstract $item */
                 $item = $this->getNewEmptyItem();
                 if ($this->getIdFieldName()) {
                     $item->setIdFieldName($this->getIdFieldName());
                 }
                 $item->addData($row);
+                $item->setOrigData();
+                $item->setLastSavedData();
                 $this->addItem($item);
             }
         }
```

since it should be on a forked branch instead of the main repo (and I'm cleaning the branches of the official repo), sorry for the inconvenience!

**Pull request closed without merging**